### DISCUSSION TITLE
Revisions are not returned

### DISCUSSION THREAD

**Discussion by joebailey26** [wp-graphql/wp-graphql#2957] [Q&A] [answered]
I can see post revisions in the database and on the page in WordPress, but they are not returned in WP GraphQL using the example in the docs.
This is my query:
```
{
  page(id: 8655, idType: DATABASE_ID) {
    title
    revisions(first: 1, where: {orderby: {field: DATE, order: DESC}}) {
      edges {
        node {
          id
          title
          date
          content
        }
      }
    }
  }
}
```
Returns:
```
{
  "data": {
    "page": {
      "title": "test preview",
      "revisions": {
        "edges": []
      }
    }
  }
}
```

I'm on v1.16.0.
I've tested this on two different WordPress sites, on different post types, with different queries, such as root revision query, or `postBy`. WP-GraphQL just does not seem to recognize revisions.

**Comment by rodrigo-arias** [comment]
@joebailey26 This has been resolved in WPGraphQL [v1.18](https://github.com/wp-graphql/wp-graphql/releases/tag/v1.18.0) with https://github.com/wp-graphql/wp-graphql/pull/2937 so the two queries below should return the same.

````
query GET_REVISION {
  revision: page(id: "test", idType: URI) {
    title
    revisions(first: 1, where: {orderby: {field: DATE, order: DESC}}) {
      edges {
        node {
          id
          title
          date
          content(format: RENDERED)
        }
      }
    }
  }
  preview: page(id: "test", idType: URI, asPreview: true) {
    id
    title
    date
    content(format: RENDERED)
  }
}
````

Side note: Make sure the requests are authenticated to get the expected result.

**Reply by joebailey26** [reply]
Thanks @rodrigo-arias, I'm on v1.20 and I still have an issue with no revisions being returned even when using the query in your comment.

```
query GET_REVISION {
  revision: page(id: "accreditations", idType: URI) {
    title
    revisions(first: 1, where: {orderby: {field: DATE, order: DESC}}) {
      edges {
        node {
          id
          title
          date
          content(format: RENDERED)
        }
      }
    }
  }
  page: page(id: "accreditations", idType: URI, asPreview: true) {
    id
    title
    date
    content(format: RENDERED)
  }
}
```

```
{
  "data": {
    "revision": {
      "title": "Accreditations",
      "revisions": {
        "edges": []
      }
    },
    "page": null
  },
  "extensions": {
    "debug": [
      {
        "type": "DEBUG_LOGS_INACTIVE",
        "message": "GraphQL Debug logging is not active. To see debug logs, GRAPHQL_DEBUG must be enabled."
      }
    ]
  }
}
```

![image](https://github.com/wp-graphql/wp-graphql/assets/18091947/d57aac0e-ed53-4d4a-8dc0-2ce6566bfd2f)

**Reply by rodrigo-arias** [reply, marked as answer]
@joebailey26 Make sure your requests are authenticated as WPGraphQL follows the same rules as WordPress in terms of roles and permission.

You can do it from GraphiQL with the following toggle
![CleanShot 2024-02-02 at 13 55 23](https://github.com/wp-graphql/wp-graphql/assets/11416255/b75fc868-770c-444f-a9a7-2e6355c64860)

**Reply by joebailey26** [reply]
Thanks so much @rodrigo-arias I feel silly missing that!

**Reply by rodrigo-arias** [reply]
No problem, I think there are plans to document it better https://github.com/wp-graphql/wp-graphql/issues/3021
